### ISSUE TITLE
Alert always update Active Since date to current time and cant fire

### ISSUE THREAD

**Issue by Teriand** [prometheus/prometheus#4652]
## Bug Report
**What did you do?**
After upgrade 2.3.2 ->2.4.2
Alert always update Active Since date to current time and cant fire:

![image](https://user-images.githubusercontent.com/26705069/46004770-5c510b80-c0bc-11e8-85c0-8d092581efe0.png)
ex on screen
alert: IgnoreAlert
expr: vector(1)
for: 10s

cant fire. Before update i use him for tesing and it fire work.
"Active Since" always set after refresh page to current time.
 or
 curl http://localhost:9090/api/v1/alerts

,{"labels":{"alertname":"IgnoreAlert","grafana":"http://prometheus-01:3000?from=1537865257060\u0026to=1537868857060","severity":"warn","type":"test","zone":"test"},"annotations":{"description":"Daily alert test description","summary":"Daily alert test summary http://prometheus-01:3000/d/000000011/windows-node-test-stands?refresh=10s\u0026orgId=1\u0026var-interval=60s\u0026var-server=auto-fna-62\u0026from=now-3h\u0026to=now"},"state":"pending","activeAt":"2018-09-25T12:17:37.060655984+03:00","value":1}]}}

**What did you expect to see?**
Change Pending to Fire state

**What did you see instead? Under which circumstances?**
Persist Pending state

**Environment**

* System information:

Linux 4.1.12-61.1.18.el7uek.x86_64 x86_64

* Prometheus version:
prometheus, version 2.4.2 (branch: HEAD, revision: c305ffaa092e94e9d2dbbddf8226c4813b1190a0)
  build user:       root@dcde2b74c858
  build date:       20180921-07:22:29
  go version:       go1.10.3

* Prometheus configuration file:
```
global:
  scrape_interval:     15s
  evaluation_interval: 15s
  scrape_timeout: 15s
alerting:
  alertmanagers:
  - static_configs:
    - targets: ['prometheus-01:9093']
rule_files:
   - "test-alert.yml"
scrape_configs:
  - job_name: 'prometheus'
    static_configs:
      - targets: ['prometheus-01:9090']
    metric_relabel_configs:
      - source_labels: [ __name__ ]
        regex: 'go_(.*)'
        action: drop
```

**Comment by simonpasquier** [comment]
This is because your `grafana` label value is changing everytime Prometheus evaluates the alert. It was a bug (https://github.com/prometheus/prometheus/issues/4169) in the alerting code and it has been fixed in 2.4.x. You should store this value as an annotation instead.

**Comment by Teriand** [comment]
but #4169 closed and #4500 merget, next #4571 track it ? 

ps
move grafana from labels to annotations work fine. thx

**Comment by simonpasquier** [comment]
#4571 can be closed too if it is really a duplicate of #4169.

**Comment by Teriand** [comment]
and mb open my issue to track fix alerting code in 2.4.x?

**Comment by simonpasquier** [comment]
I'm not sure to understand your last question but there's no bug to be fixed. The alert's identity is defined by its labels. If the labels aren't constant over time then they are different alerts.

**Comment by Teriand** [comment]
ok, i mistake u message: " It was a bug (#4169) in the alerting code and it has been fixed in 2.4.x."
I change all my alert-rule files for annotatons.
thx.
